internal/inspect/detectors/plumber.go

```diff
@@ -3,38 +3,151 @@ package detectors
+var possiblePlumberEntrypoints = []string{"plumber.R", "entrypoint.R"}
+var possiblePlumberServerFiles = []string{"_server.yml", "_server.yaml"}
+
+func NewPlumberDetector(log logging.Logger) *PlumberDetector {
 	return &PlumberDetector{
 		inferenceHelper: defaultInferenceHelper{},
+		log:             log,
 		executor:        executor.NewExecutor(),
 	}
 }
 
+func (d *PlumberDetector) isSupportedEntrypoint(entrypoint util.RelativePath) bool {
+	return entrypoint.Ext() == ".R" || entrypoint.Ext() == ".yml" || entrypoint.Ext() == ".yaml"
+}
+
 func (d *PlumberDetector) InferType(base util.AbsolutePath, entrypoint util.RelativePath) ([]*config.Config, error) {
 	requiredEntrypoint := entrypoint.String()
 	if requiredEntrypoint != "" {
 		// Optimization: skip inspection if there's a specified entrypoint
 		// and it's not one of ours.
-		if entrypoint.Ext() != ".R" {
+		if !d.isSupportedEntrypoint(entrypoint) {
 			return nil, nil
 		}
 	}
 	var configs []*config.Config
-	// rsconnect looks for these two specific entrypointPath filenames
-	possibleEntrypoints := []string{"plumber.R", "entrypoint.R"}
 
-	for _, relEntrypoint := range possibleEntrypoints {
+	configByServerfile, err := d.inferByServerFile(base, entrypoint)
+	if err != nil {
+		return nil, err
+	}
+
+	if configByServerfile != nil {
+		configs = append(configs, configByServerfile)
+		return configs, nil
+	}
+
+	configByEntrypoint, err := d.inferByEntrypoint(base, requiredEntrypoint)
+	if err != nil {
+		return nil, err
+	}
+
+	if configByEntrypoint != nil {
+		configs = append(configs, configByEntrypoint)
+		return configs, nil
+	}
+
+	return configs, nil
+}
+
+func (d *PlumberDetector) includeServerYmlFiles(cfg *config.Config, metadata plumberServerMetadata) {
+	if metadata.Constructor != "" {
+		d.log.Info("Plumber project detection, including constructor file.", "constructor", metadata.Constructor)
+		cfg.Files = append(cfg.Files, fmt.Sprint("/", metadata.Constructor))
+	}
+
+	routesString, ok := metadata.Routes.(string)
+	if ok && routesString != "" {
+		d.log.Info("Plumber project detection, including routes file.", "routes_file", routesString)
+		cfg.Files = append(cfg.Files, fmt.Sprint("/", routesString))
+	}
+
+	routesList, ok := metadata.Routes.([]any)
+	if ok && routesList != nil && len(routesList) > 0 {
+		d.log.Info("Plumber project detection, routes is a list.")
+		for _, routeFile := range routesList {
+			routesFileString, ok := routeFile.(string)
+			if ok && routesFileString != "" {
+				d.log.Info("Plumber project detection, including routes file.", "routes_file", routesFileString)
+				cfg.Files = append(cfg.Files, fmt.Sprint("/", routesFileString))
+			}
+		}
+	}
+}
+
+func (d *PlumberDetector) inferByServerFile(base util.AbsolutePath, entrypoint util.RelativePath) (*config.Config, error) {
+	for _, serverFile := range possiblePlumberServerFiles {
+		// Does a server file exist?
+		serverFilePath := base.Join(serverFile)
```

Comment:
If they have both `_server.yml` and `_server.yaml` it looks like this will use just the first one, yes? Hopefully this is an unlikely scenario, but in rsconnect we error if we detect both.

Reply:
Yes, this will only use the first one that is found. With this, I'm not convinced it is worth checking if both are present and erroring out. Still happy to re-consider if there is something you think I'm missing.